### PR TITLE
fix: out of mem after Requesting tags to be replaced

### LINKED ISSUE #408: Invalid string length on syncronisation. Developer Console pauses before potential out-of-memory
Hey,
I'm getting the following error after I press the sync button:
image
To narrow down the error, I created only one card:
`STARTI [Basic] This is a test. Back: Test successful! ENDI `
It gets generated:
`STARTI [Basic] This is a test. Back: Test successful! <!--ID: 1677148389236--> ENDI`
However, if I press the sync button again, Obsidian dies with the error stated above.
Any help appreciated.
Thanks

### PR DESCRIPTION
**Pull request by MisterPresident**
Fix #408 by skipping empty files.

### CODE REVIEW

**Comment by MisterPresident** [comment]
#428

**Comment by IOSure** [comment]
#408

**Comment by MisterPresident** [comment]
> Thanks! Do you think this is something that would fix #446 ?

Yes I think so.

File: src/files-manager.ts [hunk 1]
```diff
@@ -297,7 +297,10 @@ export class FileManager {
         temp = []
         console.info("Requesting tags to be replaced...")
         for (let file of this.ownFiles) {
-            temp.push(file.getClearTags())
+             let rem = file.getClearTags()
+             if(rem.params.notes.length) {
+                  temp.push(rem))
```
Comment (ShootingKing-AM) [inline src/files-manager.ts:302]: There is a extra ')' at the end of the line, does this fix really work ?

Comment (MisterPresident) [reply src/files-manager.ts:302]: I don't know how it got there. It seems i committed nonsense. I am sorry, I'm going to fix it when I am at home.

**Comment by MisterPresident** [comment]
> Cant reproduce the linked issues on my end, but it also makes sense to not scan empty files. Tests for this PR are done in #494 Thanks for the PR.

Thank you for merging.

How much .md files does your vault have? I guess it depends on the number of files. >1000?

**Comment by ShootingKing-AM** [comment]
> > Cant reproduce the linked issues on my end, but it also makes sense to not scan empty files. Tests for this PR are done in #494 Thanks for the PR.
> 
> Thank you for merging.
> 
> How much .md files does your vault have? I guess it depends on the number of files. >1000?

Sorry, didnt see your message. Maybe we can write a Test for this. I definately will not have 1000 files

**Pull request merged**